libs/ledger-live-common/src/hw/deviceAccess.ts

```diff
@@ -175,6 +175,11 @@ export const withDevice =
       };
     });
 
+export const withDevicePromise = <T>(
+  deviceId: string,
+  fn: (transport: Transport) => Observable<T>
+): Promise<T> => withDevice(deviceId)((transport) => fn(transport)).toPromise();
```

Comment:
this is IMO not something we want to generalize, this pattern prevent the ability to cancel/interrupt the flow which the UI usually can do (the caller of withDevicePromise can no longer interrupts). Like we instead would like to kill many places where we needed it and convert them to observable. makes me wonder if that signer step should be a promise or actually an observable.
in case of bitcoin, we for instance expect the signer to emit many events, so we can precisely have these steps:

0%...10%...50%... sign on device appears ... signed ... 80% ... 100%

Reply:
Ok, I get your point.
The issue then is that de `getAddress` methods are not reactive. We need also to change their signature, otherwise it will never working as you propose.
WDYT?

Reply:
No idea, not quite familiar with all this. @gre any though on @sprohaszka-ledger last comment?